src/query/storages/fuse/src/table_functions/fuse_encodings/fuse_encoding.rs

```diff
@@ -0,0 +1,222 @@
+        let mut uncompressed_size = Vec::with_capacity(pages_info.len());
+        let mut l1 = StringColumnBuilder::with_capacity(pages_info.len(), pages_info.len());
+        let mut l2 = NullableColumnBuilder::<StringType>::with_capacity(pages_info.len(), &[]);
+        for p in pages_info {
+            validity_size.push(p.validity_size);
+            compressed_size.push(p.compressed_size);
+            uncompressed_size.push(p.uncompressed_size);
+            l1.put_slice(encoding_to_string(&p.body).as_bytes());
+            l1.commit_row();
+            let l2_encoding = match &p.body {
+                PageBody::Dict(dict) => Some(encoding_to_string(&dict.indices.body)),
+                PageBody::Freq(freq) => freq
+                    .exceptions
+                    .as_ref()
+                    .map(|e| encoding_to_string(&e.body)),
+                _ => None,
+            };
+            if let Some(l2_encoding) = l2_encoding {
+                l2.push(l2_encoding.as_bytes());
+            } else {
+                l2.push_null();
+            }
+        }
+        Ok(DataBlock::new(
+            vec![
+                BlockEntry::new(
+                    DataType::Nullable(Box::new(DataType::Number(NumberDataType::UInt32))),
+                    Value::Column(UInt32Type::from_opt_data(validity_size)),
+                ),
+                BlockEntry::new(
+                    DataType::Number(NumberDataType::UInt32),
+                    Value::Column(UInt32Type::from_data(compressed_size)),
+                ),
+                BlockEntry::new(
+                    DataType::Number(NumberDataType::UInt32),
+                    Value::Column(UInt32Type::from_data(uncompressed_size)),
+                ),
+                BlockEntry::new(DataType::String, Value::Column(Column::String(l1.build()))),
+                BlockEntry::new(
+                    DataType::Nullable(Box::new(DataType::String)),
+                    Value::Column(Column::Nullable(Box::new(l2.build().upcast()))),
+                ),
+            ],
+            num_row,
+        ))
+    }
+
+    pub fn schema() -> Arc<TableSchema> {
+        TableSchemaRefExt::create(vec![
+            TableField::new(
+                "validity_size",
+                TableDataType::Nullable(Box::new(TableDataType::Number(NumberDataType::UInt32))),
+            ),
+            TableField::new(
+                "compressed_size",
+                TableDataType::Number(NumberDataType::UInt32),
+            ),
+            TableField::new(
+                "uncompressed_size",
+                TableDataType::Number(NumberDataType::UInt32),
+            ),
+            TableField::new("level_one", TableDataType::String),
+            TableField::new(
+                "level_two",
+                TableDataType::Nullable(Box::new(TableDataType::String)),
+            ),
+        ])
+    }
+}
+
+fn encoding_to_string(page_body: &PageBody) -> String {
```

Comment:
we can move this function to `strawboat` to make it more reusable.

Reply:
Good idea, I will do this in next PR